sklearn/metrics/tests/test_pairwise.py

```diff
@@ -438,7 +439,93 @@ def test_cosine_distances():
     assert_true(np.all(D <= 2.))
 
 
-# Paired distances
+def test_gower_distances():
+    # Check the pairwise Gower distances computation
+
+    # Zero distance expected
+    X = np.array([[1, 4141.22, False, 'ABC'],
+                  [1, 4141.22, False, 'ABC']], dtype=object)
+
+    D = gower_distances(X)
+
+    assert_array_almost_equal(D, np.array([[0.0, 0.0],
+                                           [0.0, 0.0]]))
+
+    X = np.array([['M', False, 222.22, 1],
+                  ['F', True, 333.22, 2],
+                  ['M', True, 1934.0, 4],
+                  [None, None, np.nan, np.nan]], dtype=object)
+
+    D = gower_distances(X)
+
+    D_expected = np.array([[0.0, 0.599545, 0.75, np.nan],
```

Comment:
Where do these distances come from? How do I know they are accurate? If they can be calculated trivially then you should show the working in the test. If they come from a reference paper, please provide the reference.